File: opensource/kernel/src/main/java/com/sos/scheduler/engine/eventbus/EventBus.java
package com.sos.scheduler.engine.eventbus;

import com.sos.scheduler.engine.data.event.Event;

public interface EventBus {
    void registerAnnotated(EventHandlerAnnotated o);
    void unregisterAnnotated(EventHandlerAnnotated o);
    void register(EventSubscription o);
    void unregister(EventSubscription o);
    boolean isSubscribed();
    void publish(Event e);
}

File: opensource/kernel/src/main/scala/com/sos/scheduler/engine/eventbus/EventBus.scala
package com.sos.scheduler.engine.eventbus

import com.sos.scheduler.engine.data.event.Event

/**
 * @author Joacim Zschimmer
 */
trait EventBus {

  def registerAnnotated(o: EventHandlerAnnotated): Unit

  def unregisterAnnotated(o: EventHandlerAnnotated): Unit

  def register(o: EventSubscription): Unit

  def unregister(o: EventSubscription): Unit

  def publish(e: Event): Unit
}

